libs/platform/core/src/services/token-resolver/default-token-resolver.service.ts

```
@@ -9,34 +9,63 @@ import {
 } from './token-resolver.service';
 
 const tokenRE = /^[A-Z_-]+\.(!?)[A-Z_-]+$/;
+const conditionalTokenRE = /\|\|/;
 
 export class DefaultTokenService implements TokenResolver {
   protected resolvers = new Map<string, TokenResourceResolver>();
   protected injector = inject(INJECTOR);
 
   resolveToken(token: string): ResolvedToken {
+    if (this.isConditionalToken(token)) {
+      return this.resolveConditionalToken(token);
+    }
     if (!this.isToken(token)) {
       return of(token);
     }
+    return this.resolveSimpleToken(token);
+  }
+
+  protected resolveConditionalToken(token: string): ResolvedToken {
+    const tokensToResolve = token
+      .split('||')
+      .map((token) => token.trim())
+      .filter(
+        (token) => this.isToken(token) && this.getResolver(token.split('.')[0])
+      );
+
+    if (!tokensToResolve.length) {
```

This is a flaky behaviour, I believe better to throw an error instead, to make it clear that tokens are invalid. Because if we return false for invalid tokens will not be so easy to debug it. Or at least console.error, for each invalid token